src/build.cc

```diff
@@ -130,6 +131,18 @@ void BuildStatus::BuildEdgeFinished(Edge* edge,
 
   if (!edge->use_console())
     PrintStatus(edge, kEdgeFinished);
+  if (printer_.is_smart_terminal()) {
+    int oldest_start = INT_MAX;
+    Edge* oldest = NULL;
+    for (i = running_edges_.begin(); i != running_edges_.end(); i++) {
+      if (i->second < oldest_start) {
+        oldest_start = i->second;
+        oldest = i->first;
+      }
+    }
+    if (oldest)
+      PrintStatus(oldest, kEdgeRunning);
```

Comment:
This seems like the wrong place, if the finished edge has any output or fails this will overwrite the status line for the finished edge with the status line for a still running edge and then follow it with the output of the finished edge.  I think this whole block has to move to the end of the function.